test/cluster/security_test.go

```diff
@@ -0,0 +1,336 @@
+		require.NoError(t, err)
+		require.NotNil(t, rcpt)
+		require.True(t, txID != "")
+		require.True(t, len(rcpt.GetHeader().GetValidationInfo()) > 0)
+		require.Equal(t, types.Flag_VALID, rcpt.Header.ValidationInfo[rcpt.TxIndex].Flag)
+		txsCount++
+		clusterLogger.Info("tx submitted: "+txID+", ", rcpt)
+	}
+
+	require.Eventually(t, func() bool {
+		return c.AgreedHeight(t, uint64(txsCount+1), 0, 1, 2)
+	}, 30*time.Second, 100*time.Millisecond)
+
+	follower1 := (leaderIndex + 1) % 3
+	follower2 := (leaderIndex + 2) % 3
+
+	require.NoError(t, c.ShutdownServer(c.Servers[leaderIndex]))
+
+	//find new leader
+	newLeaderIndex := -1
+	require.Eventually(t, func() bool {
+		newLeaderIndex = c.AgreedLeader(t, follower1, follower2)
+		return newLeaderIndex >= 0
+	}, 30*time.Second, 100*time.Millisecond)
+
+	//submit txs with data size > SnapshotIntervalSize
+	for _, key := range keys {
+		txID, rcpt, err := c.Servers[newLeaderIndex].WriteDataTx(t, worldstate.DefaultDBName, strconv.Itoa(key+100), data)
+		clusterLogger.Info("key-", key+100)
+		require.NoError(t, err)
+		require.NotNil(t, rcpt)
+		require.True(t, txID != "")
+		require.True(t, len(rcpt.GetHeader().GetValidationInfo()) > 0)
+		require.Equal(t, types.Flag_VALID, rcpt.Header.ValidationInfo[rcpt.TxIndex].Flag)
+		txsCount++
+		clusterLogger.Info("tx submitted: "+txID+", ", rcpt)
+	}
+
+	require.Eventually(t, func() bool {
+		return c.AgreedHeight(t, uint64(txsCount+1), follower1, follower2)
+	}, 30*time.Second, 100*time.Millisecond)
+
+	//restart the victim
+	require.NoError(t, c.StartServer(c.Servers[leaderIndex]))
+
+	//make sure the server is in sync with previous txs
+	require.Eventually(t, func() bool {
+		return c.AgreedHeight(t, uint64(txsCount+1), 0, 1, 2)
+	}, 30*time.Second, 100*time.Millisecond)
+
+	keys = createKeys(200)
+	var dataEnv *types.GetDataResponseEnvelope
+	for _, key := range keys {
+		require.Eventually(t, func() bool {
+			dataEnv, err = c.Servers[follower1].QueryData(t, worldstate.DefaultDBName, strconv.Itoa(key), "admin")
+			return dataEnv != nil && dataEnv.GetResponse().GetValue() != nil && err == nil
+		}, 30*time.Second, 100*time.Millisecond)
+		dataResp := dataEnv.GetResponse().GetValue()
+		require.Equal(t, dataResp, data)
+	}
+
+	newLeaderIndex = -1
+	require.Eventually(t, func() bool {
+		newLeaderIndex = c.AgreedLeader(t, 0, 1, 2)
+		return newLeaderIndex >= 0
+	}, 30*time.Second, 100*time.Millisecond)
+
+	require.NoError(t, c.Shutdown())
+
+	snapList = replication.ListSnapshots(clusterLogger, filepath.Join(c.Servers[newLeaderIndex].ConfigDir(), "etcdraft", "snap"))
+	require.NotEqual(t, 0, len(snapList))
+	clusterLogger.Info("Snap list: ", snapList)
+}
+
+// Scenario:
+// - start 3 servers in a cluster with TLS enabled.
+// - stop the leader.
+// - wait for new server to be the leader.
+// - the leader submit a tx.
+// - repeat until each server was a leader at least once.
+// - make sure each node submitted a valid tx as a leader.
```

Comment:
You can drop this scenario as it does not add much to the security testing.

Instead, add a scenario in which we test negative cases: 
- a server with TLS disabled cannot connect to the cluster
- a server with TLS enabled but a bad certificate either will not start or won't connect
- a server with TLS enabled but a key-cert pair not from the cluster's CA will not connect
etc.

That is, start a cluster with 3 servers, in which one is misconfigured as indicated above, start the cluster, make sure the correct 2 do not connect to the misconfigured one.

You can do all the negative tests in another PR, "TLS negative tests", this way we'll move faster.